lib/net.c

```diff
@@ -42,17 +43,35 @@
 
 static struct net_config config;
 
-/*
- * XXX The secret key of the RSS hash must be random
- * in order to avoid hackers to know it.
- */
-uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN] = {
-	0x6d, 0x5a, 0x56, 0xda, 0x25, 0x5b, 0x0e, 0xc2,
-	0x41, 0x67, 0x25, 0x3d, 0x43, 0xa3, 0x8f, 0xb0,
-	0xd0, 0xca, 0x2b, 0xcb, 0xae, 0x7b, 0x30, 0xb4,
-	0x77, 0xcb, 0x2d, 0xa3, 0x80, 0x30, 0xf2, 0x0c,
-	0x6a, 0x42, 0xb7, 0x3b, 0xbe, 0xac, 0x01, 0xfa,
-};
+uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN];
+
+/* Function to randomize the default RSS key. */
+static void
+randomize_rss_key(void)
+{
+	uint8_t i, temp_key, temp_count;
+	uint8_t final_set_count = 0;
+	ssize_t number_of_bytes;
+
+	number_of_bytes = getrandom(default_rss_key, GATEKEEPER_RSS_KEY_LEN, 0);
```

Comment:
Did you try to compile the code with the modification? It seems that the ``getrandom()`` is a system call, and it cannot be found in C. If you did have the issue, please refer to https://stackoverflow.com/questions/30800331/getrandom-syscall-in-c-not-found

Reply:
The final parameter of ``getrandom()`` indicates the way it works. Please add a comment to explain your choice, that is why ``0``. Please make sure the choice meets with our need, and ensure the randomness as much as possible.

Reply:
The code compiles fine on my Ubuntu 17.10 with kernel version 4.13.0 and glibc version 2.26. To test the code, I tried compiling the same code on Ubuntu 16.04 with kernel version 4.4.0 and glibc version 2.23 but it fails with the following error
`/home/sudeep/gatekeeper/lib/net.c:24:24: fatal error: sys/random.h: No such file or directory
`

The `getrandom()` function requires glibc version >= 2.25 and kernel version >= 3.17

Reply:
So, I would recommend to use ``syscall()`` to avoid the compatibility issue.